lib/config/defaults.js

```diff
@@ -602,102 +644,109 @@ const applyOptimizationDefaults = (
 };
 
 /**
- * @param {ResolveOptions} resolve options
  * @param {Object} options options
  * @param {boolean} options.cache is cache enable
  * @param {string} options.context build context
  * @param {boolean} options.mjs is mjs enabled
  * @param {boolean} options.webTarget is a web-like target
  * @param {Target} options.target target
  * @param {Mode} options.mode mode
- * @returns {void}
+ * @returns {ResolveOptions} resolve options
  */
-const applyResolveDefaults = (
-	resolve,
-	{ cache, context, mjs, webTarget, target, mode }
-) => {
-	D(resolve, "cache", cache);
-	D(resolve, "modules", ["node_modules"]);
-	F(resolve, "conditionNames", () => {
-		/** @type {string[]} */
-		const conditions = ["webpack"];
+const getResolveDefaults = ({
+	cache,
+	context,
+	mjs,
+	webTarget,
+	target,
+	mode
+}) => {
+	/** @type {string[]} */
+	const conditions = ["webpack"];
+
+	conditions.push(mode === "development" ? "development" : "production");
+
+	switch (target) {
+		case "webworker":
+			conditions.push("worker");
+			break;
+		case "node":
+		case "async-node":
+		case "node-webkit":
+			conditions.push("node");
+			break;
+		case "electron-main":
+		case "electron-preload":
+			conditions.push("node");
+			conditions.push("electron");
+			break;
+		case "electron-renderer":
+			conditions.push("electron");
+			break;
+	}
```

Comment:
Do we really need this stuff for non JS types? For stylesheets we don't need `worker`, `node`, `electron` conditions

Reply:
It's about the environment the stuff runs in. User could use different styles depending on if the code is compiled for electron or web (hopefully noone never does that, but that's how it works)

Reply:
Make sense, but sounds terrible :smile: